### DISCUSSION TITLE
PowerOnState isn't working fine

### DISCUSSION THREAD

**Discussion by Srjosep** [arendst/Tasmota#12738] [Q&A] [answered]
I have an ESP32 wroom tasmotized and plugged to a 8 relays board. It works fine, I can turn every relay on and off by Tasmota frontend toogle button or by MQTT with HA.

But even I have PowerOnState set to 1, after a power off, some relays keeps off, maybe 2 or 3 go to on.
What could be the problem here?

**Comment by cociweb** [comment, marked as answer]
It seems, that is a power supply issue. It depends on the relay specification, but usually 15-30 mA nominal current needs to keep a 12V monostable relay in one position for each channel. (significantly more current need at the moment of the switching). If your relay operates on lower voltage the nominal current can reach 100 mA for each.  if you start up your device it will be summarized, so it can easily reach 1-2A at the moment of the startup. 
So based on your proportion: 2/8 you may need 4 times more current. for the relay board. Please check your board specification before you accidentally  destroy it....

**Reply by cociweb** [reply]
Alternativelly You can use rules to delay the channels from each other for a few moment: https://tasmota.github.io/docs/Rules/

**Reply by Srjosep** [reply]
I tried all this, even tried with the relays board disconnected from the ESP32, and the problem si the same.  The power supply is 2A, more than enought for an ESP32 wroom. It seems I don't know why, my setup is deleted every time I unplug the device or restart it.

**Reply by barbudor** [reply]
Is it Tasmota not setting the relay as per PowerOnState or is it the configuration of PowerOnState that disappear after a power-cycle

In the later case, You may have disabled saving ettings to flash like using `SaveData 0`


Note that a restart is not a power-off and PowerOnState do not applies to a restart. Only to a power-on.
On a restart, relay are not supposed to change but this also can be impacted by `SetOption 0`.

**Reply by Srjosep** [reply]
the configuration of PowerOnState is what disappear after a power-cycle. And timers, and somethings more... but don't turn to default completely. SetOption0 is on.

**Reply by barbudor** [reply]
Very strange. Can you please
- set 'Weblog 4'
- set 'PowerOnState_x_ _n_' at your liking
- wait 10 seconds
- copy/paste here the content of the Weblog from 'Weblog 4' until the end
- power cycle the device 
- after restart type 'PowerOnState' to display the values
- copy paste here the Weblog from 1st line after reboot until the 'poweronstate' result

Please don't do any screenshots but a real text-copy-paste of the logs using the code marks here (the `<>` button above)

**Reply by barbudor** [reply]
Sorry I didn't meant 'restart 1' but power-cycle the device

**Reply by Srjosep** [reply]
Before power cycle
`17:35:42.733 CMD:  weblog 4
17:35:42.745 MQT: controlador_acuario_marino/stat/RESULT = {"WebLog":4}
17:35:43.556 CFG: Saved, Count 27, Bytes 4096
17:35:50.593 WIF: Checking connection...
17:35:51.815 MQT: controlador_acuario_marino/tele/STATE = {"Time":"2021-08-15T17:35:51","Uptime":"0T00:10:15","UptimeSec":615,"Heap":189,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":1,"POWER1":"ON","POWER2":"ON","POWER3":"ON","POWER4":"ON","POWER5":"OFF","POWER6":"ON","POWER7":"ON","POWER8":"OFF","Wifi":{"AP":1,"SSId":"JcJ","BSSId":"54:67:51:58:72:9D","Channel":1,"Mode":"11n","RSSI":38,"Signal":-81,"LinkCount":1,"Downtime":"0T00:00:07"}}
17:35:51.838 MQT: controlador_acuario_marino/tele/SENSOR = {"Time":"2021-08-15T17:35:51","ESP32":{"Temperature":60.6},"TempUnit":"C"}
17:36:52.901 CMD: PowerOnState 1
17:36:52.904 SRC: WebConsole from 192.168.1.49
17:36:52.908 CMD: Grp 0, Cmnd 'POWERONSTATE', Idx 1, Len 1, Data '1'
17:36:52.922 MQT: controlador_acuario_marino/stat/RESULT = {"PowerOnState":1}
17:36:53.922 CFG: Saved, Count 28, Bytes 4096
17:37:10.816 WIF: Checking connection...`

After the power cycle
`00:00:00.002 HDW: ESP32-D0WDQ6
00:00:00.048 UFS: FlashFS mounted with 304 kB free
00:00:00.125 CFG: Loaded from File, Count 19
00:00:00.139 QPC: Count 1
00:00:00.160 BRY: Berry initialized, RAM used=1760
00:00:00.162 BRY: no 'preinit.be'
00:00:00.168 Project tasmota Controlador_acuario_marino Version 9.5.0.2(tasmota)-1_0_7_1(2021-07-05T17:11:58)
00:00:00.172 BRY: no 'autoexec.be'
00:00:00.996 WIF: Connecting to AP1 JcJ Channel 11 BSSId 54:67:51:58:72:9D in mode 11n as controlador_acuario_marino-5144...
00:00:06.843 QPC: Reset
00:00:06.927 WIF: Connected
00:00:07.140 HTP: Web server active on controlador_acuario_marino-5144 with IP address 192.168.1.15
17:39:02.472 MQT: Attempting connection...
17:39:02.589 MQT: Connected
17:39:02.604 MQT: controlador_acuario_marino/tele/LWT = Online (retained)
17:39:02.621 MQT: controlador_acuario_marino/cmnd/POWER = 
17:39:02.658 MQT: controlador_acuario_marino/tele/INFO1 = {"Info1":{"Module":"ESP32-DevKit","Version":"9.5.0.2(tasmota)","FallbackTopic":"cmnd/Controlador acuario Marino_fb/","GroupTopic":"tasmotas/cmnd/"}}
17:39:02.673 MQT: controlador_acuario_marino/tele/INFO2 = {"Info2":{"WebServerMode":"Admin","Hostname":"controlador_acuario_marino-5144","IPAddress":"192.168.1.15"}}
17:39:02.687 MQT: controlador_acuario_marino/tele/INFO3 = {"Info3":{"RestartReason":"RTC Watch dog reset digital core and rtc module"}}
17:39:02.699 MQT: controlador_acuario_marino/stat/RESULT = {"POWER1":"OFF"}
17:39:02.707 MQT: controlador_acuario_marino/stat/POWER1 = OFF
17:39:02.713 MQT: controlador_acuario_marino/stat/RESULT = {"POWER2":"OFF"}
17:39:02.719 MQT: controlador_acuario_marino/stat/POWER2 = OFF
17:39:02.726 MQT: controlador_acuario_marino/stat/RESULT = {"POWER3":"OFF"}
17:39:02.732 MQT: controlador_acuario_marino/stat/POWER3 = OFF
17:39:02.738 MQT: controlador_acuario_marino/stat/RESULT = {"POWER4":"OFF"}
17:39:02.744 MQT: controlador_acuario_marino/stat/POWER4 = OFF
17:39:02.750 MQT: controlador_acuario_marino/stat/RESULT = {"POWER5":"OFF"}
17:39:02.756 MQT: controlador_acuario_marino/stat/POWER5 = OFF
17:39:02.762 MQT: controlador_acuario_marino/stat/RESULT = {"POWER6":"OFF"}
17:39:02.768 MQT: controlador_acuario_marino/stat/POWER6 = OFF
17:39:02.774 MQT: controlador_acuario_marino/stat/RESULT = {"POWER7":"OFF"}
17:39:02.780 MQT: controlador_acuario_marino/stat/POWER7 = OFF
17:39:02.787 MQT: controlador_acuario_marino/stat/RESULT = {"POWER8":"OFF"}
17:39:02.793 MQT: controlador_acuario_marino/stat/POWER8 = OFF
17:39:04.923 MQT: controlador_acuario_marino/stat/RESULT = {"Time":"2021-08-15T17:39:04","Uptime":"0T00:00:12","UptimeSec":12,"Heap":191,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":307,"MqttCount":1,"POWER1":"OFF","POWER2":"OFF","POWER3":"OFF","POWER4":"OFF","POWER5":"OFF","POWER6":"OFF","POWER7":"OFF","POWER8":"OFF","Wifi":{"AP":1,"SSId":"JcJ","BSSId":"54:67:51:58:72:9D","Channel":1,"Mode":"11n","RSSI":46,"Signal":-77,"LinkCount":1,"Downtime":"0T00:00:08"}}
17:39:04.984 MQT: controlador_acuario_marino/stat/STATUS10 = {"StatusSNS":{"Time":"2021-08-15T17:39:04","ESP32":{"Temperature":61.7},"TempUnit":"C"}}
17:39:07.818 MQT: controlador_acuario_marino/tele/STATE = {"Time":"2021-08-15T17:39:07","Uptime":"0T00:00:15","UptimeSec":15,"Heap":192,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":76,"MqttCount":1,"POWER1":"OFF","POWER2":"OFF","POWER3":"OFF","POWER4":"OFF","POWER5":"OFF","POWER6":"OFF","POWER7":"OFF","POWER8":"OFF","Wifi":{"AP":1,"SSId":"JcJ","BSSId":"54:67:51:58:72:9D","Channel":1,"Mode":"11n","RSSI":48,"Signal":-76,"LinkCount":1,"Downtime":"0T00:00:08"}}
17:39:07.863 MQT: controlador_acuario_marino/tele/SENSOR = {"Time":"2021-08-15T17:39:07","ESP32":{"Temperature":61.7},"TempUnit":"C"}`
PowerOnSate after the power cycle
`17:40:53.179 CMD: PowerOnState
17:40:53.191 MQT: controlador_acuario_marino/stat/RESULT = {"PowerOnState":3}`

weblog after the power cycle:
`17:42:09.752 CMD: weblog
17:42:09.768 MQT: controlador_acuario_marino/stat/RESULT = {"WebLog":2}`
Timers are gone too, but the module config, MQTT and wifi are here yet. I opened an issue for this, but it was closed.

**Reply by barbudor** [reply]
Thanks
Apart from you logs missing line feeds, something is wrong with saving your settings in flash
The last written block before power-cycle is 28:
```
17:36:53.922 CFG: Saved, Count 28, Bytes 4096
```
But after restart, settings are read from block 19
```
00:00:00.125 CFG: Loaded from File, Count 19
```
It should be the same like in my own log with 9.5.0.5 (from last night build):
```
19:02:36.111 CFG: Saved, Count 18, Bytes 4096
```
then:
```
00:00:00.059 CFG: Loaded from File, Count 18
```

Something may be wrong with your flash.
Can you try a full erase and reflash with latest development version through serial ?
